x-pack/platform/plugins/shared/fleet/server/services/epm/packages/install.ts

```diff
@@ -717,6 +718,22 @@ export async function installPackageWithStateMachine(options: {
       return { error: err, installType, installSource, pkgName };
     }
 
+    const excludeDataStreamTypes =
+      appContextService.getConfig()?.internal?.excludeDataStreamTypes ?? [];
+    if (!shouldIncludePackageWithDatastreamTypes(packageInfo, excludeDataStreamTypes)) {
```

Comment:
if I'm inferring this correctly, this will block packages with both logs and metrics data streams. I don't think that's what we want here. we want to block packages with *only* metrics data streams, right?

I think there was other work that done that filters out metrics data streams from being configured if user is using logs essentials and configuring a package that has a mix of data stream types. so installing/showing a mixed package is allowed.

Reply:
@jen-huang the helper `shouldIncludePackageWithDatastreamTypes`  only filter package that only have excluded datastreams so it seems working as you say